realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -81,24 +81,89 @@ public void setUp() throws Exception {
     public void executeTransaction_async() throws Throwable {
         assertEquals(0, looperThread.realm.allObjects(Owner.class).size());
 
-        looperThread.realm.executeTransaction(new Realm.Transaction() {
+        looperThread.realm.executeTransactionAsync(new Realm.Transaction() {
             @Override
             public void execute(Realm realm) {
                 Owner owner = realm.createObject(Owner.class);
                 owner.setName("Owner");
             }
-        }, new Realm.Transaction.Callback() {
+        }, new Realm.Transaction.OnSuccess() {
             @Override
             public void onSuccess() {
                 assertEquals(1, looperThread.realm.allObjects(Owner.class).size());
                 assertEquals("Owner", looperThread.realm.where(Owner.class).findFirst().getName());
                 looperThread.testComplete();
             }
+        }, new Realm.Transaction.OnError() {
 
             @Override
-            public void onError(Exception e) {
-                looperThread.testComplete();;
-                fail(e.getMessage());
+            public void onError(Throwable error) {
+                looperThread.testComplete();
```

Comment:
`looperThread.testComplete();` should be removed. `fail` will just trigger a throw.